addons/defaultstorageclass-protection/defaultstorageclass-protection.yaml

```diff
@@ -1,12 +1,11 @@
----
 apiVersion: kubeaddons.mesosphere.io/v1beta1
 kind: ClusterAddon
 metadata:
   name: defaultstorageclass-protection
   labels:
     kubeaddons.mesosphere.io/name: defaultstorageclass-protection
   annotations:
-    catalog.kubeaddons.mesosphere.io/addon-revision: "0.0.5-1"
+    catalog.kubeaddons.mesosphere.io/addon-revision: "0.0.5-2"
     appversion.kubeaddons.mesosphere.io/defaultstorageclass-protection: "0.0.5"
```

Comment:
Same questions https://github.com/mesosphere/kubernetes-base-addons/pull/863#discussion_r571233160

Reply:
Since this is a backport of what's in master, someone can contribute this in a new PR if that's desired.

I've said many times, however, that the addon-revision should not be tied to the app version. Tying them fails when we need to downgrade an app version.

Reply:
That said... the app version is wrong so there should be a new PR to address that.

Reply:
> I've said many times, however, that the addon-revision should not be tied to the app version. Tying them fails when we need to downgrade an app version.

okay so what's the need for semver for `addon-revision`? Why not just have it be `1` and iterate from there? Pretty confusing otherwise. 

Also you and @mesosphere/sig-ksphere-catalog own this project so I'm wondering why you keep repeating yourself. Why not have a meeting and come up with a system that is easier to understand and implement it? Then you can educate others on the preferred way to do it.

Reply:
Oh, wait... that wasn't merged yet... We can still make that change.

Reply:
> okay so what's the need for semver for addon-revision? Why not just have it be 1 and iterate from there? Pretty confusing otherwise.
Great question. I don't have a good reason. Semver is a standard and in all places where versions are needed, adopting that standard seems logical.